pktbuf: compiler warning on atomic_dec()

Compiling `ng_pktbuf` since 6821350 yields a compiler warning.

```
/home/martine/Repositories/RIOT-OS/RIOT/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c: In function 'ng_pktbuf_start_write':
/home/martine/Repositories/RIOT-OS/RIOT/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c:151:20: warning: passing argument 1 of 'atomic_dec' from incompatible pointer type
         atomic_dec(&pkt->users);
                    ^
In file included from /home/martine/Repositories/RIOT-OS/RIOT/core/include/mutex.h:25:0,
                 from /home/martine/Repositories/RIOT-OS/RIOT/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c:25:
/home/martine/Repositories/RIOT-OS/RIOT/core/include/atomic.h:81:19: note: expected 'struct atomic_int_t *' but argument is of type 'unsigned int *'
 static inline int atomic_dec(atomic_int_t *var)
                   ^
```

However, `atomic_int_t`'s member is signed while `pkt->users` is unsigned, which could make simple casting problematic (not to mention that `atomic_int_t::value` is volatile).

--- a/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c
+++ b/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c
@@ -30,6 +30,9 @@
 
 #include "_pktbuf_internal.h"
 
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
 static mutex_t _pktbuf_mutex = MUTEX_INIT;
 
 /* internal ng_pktbuf functions */
@@ -249,28 +252,12 @@ static ng_pktsnip_t *_pktbuf_duplicate(const ng_pktsnip_t *pkt)
         return NULL;
     }
 
+    DEBUG("pktbuf: copying %u byte from %p to %p\n", (unsigned)pkt->size,
+          pkt->data, res->data);
     memcpy(res->data, pkt->data, pkt->size);
     res->type = pkt->type;
-
-    while (pkt->next) {
-        ng_pktsnip_t *hdr = NULL;
-
-        pkt = pkt->next;
-        hdr = _pktbuf_add_unsafe(res, pkt->data, pkt->size, pkt->type);
-
-        if (hdr == NULL) {
-            do {
-                ng_pktsnip_t *next = res->next;
-
-                _pktbuf_internal_free(res->data);
-                _pktbuf_internal_free(res);
-
-                res = next;
-            } while (res);
-
-            return NULL;
-        }
-    }
+    res->next = pkt->next;
+    DEBUG("pktbuf: set res->next to %p", (void *)pkt->next);
 
     return res;
 }

--- a/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c
+++ b/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c
@@ -137,7 +137,7 @@ ng_pktsnip_t *ng_pktbuf_start_write(ng_pktsnip_t *pkt)
 
         res = _pktbuf_duplicate(pkt);
 
-        atomic_dec(&pkt->users);
+        pkt->users--;
 
         mutex_unlock(&_pktbuf_mutex);
 

